R/distro.R

```diff
@@ -25,9 +25,13 @@ distro <- function() {
   }
 
   out$id <- tolower(out$id)
-  if (grepl("bullseye", out$codename)) {
-    # debian unstable doesn't include a number but we can map from pretty name
-    out$short_version <- "11"
+  if (is.null(out$version) || out$version == "testing") {
```

Comment:
How about this instead? (I think "unstable" also might do this?)

```suggestion
  if (is.null(out$version) || identical(out$id, "debian")) {
```

Reply:
I don't *think* this will work; I added that check in, in case `lsb-release` is installed, and `version` is worked out via that.  I've pasted the outputs of `cat /etc/os-release` and `lsb_release` from each of stable/testing/unstable below to give more context for my reasoning.

In the case of testing and unstable, `version` is filled in, and so we'd never hit this logic.

I guess to check this is Debian, but also hit this logic, then maybe this would cover both the os-release and lsb-release outputs:

```suggestion
  if (identical(tolower(out$id), "debian") && (out$version %in% c("testing", "unstable") || is.null (out$id)){
```

All that said, that conditional is getting complicated.  How about I refactor this logic (that fills in the missing sections) out of `distro()` and into `os_release()` and `lsb_release()` instead @nealrichardson ?

**Stable**

```
/# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

/# lsb_release -is #  id
Debian

/# lsb_release -rs # version
11

/# lsb_release -cs # codename
bullseye
```

**Testing**
```
/# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux bookworm/sid"
NAME="Debian GNU/Linux"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

/# lsb_release -is #  id
Debian

/# lsb_release -rs # version
testing

/# lsb_release -cs # codename
bookworm
```

**Unstable**
```
/# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux bookworm/sid"
NAME="Debian GNU/Linux"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

/# lsb_release -is #  id
Debian

/# lsb_release -rs # version
unstable

/# lsb_release -cs # codename
sid
```

Reply:
I just added some new test fixtures with these. I'm ok with most any solution that makes the tests pass.